storage/tables/data-types/index.md

```diff
@@ -1,1124 +1,1170 @@
+
+### Incremental Loading
+
+When you load data incrementally, there is a difference between typed and non-typed tables. Typed tables only compare the columns of the table's primary key, while non-typed tables compare the whole row, updating rows where any value changes. This is described in detail in our documentation on [incremental loading](/storage/tables/#difference-between-tables-with-native-datatypes-and-string-tables).
+
+### Handling NULLs
+
+Columns without native types are always `VARCHAR NOT NULL`. This means you don't need to care about a specific NULL behavior. This changes with typed columns. In most databases, NULL does not equal NULL (`NULL == NULL` is not `TRUE`, but `NULL`). This breaks the incremental loading flow where columns are compared against each other.
+
+For this reason, please make sure that your primary key columns are not nullable. This is most relevant in CTAS queries, where columns are nullable by default. To work around this, specify the columns as part of the CTAS expression. For example:
+
+```sql
+CREATE TABLE "ctas_table" (
+    "id" NUMBER NOT NULL,
+    "name" VARCHAR(255) NOT NULL,
+    "created_at" TIMESTAMP_NTZ NOT NULL
+) AS SELECT * FROM "typed_table";
+```
+
+### Pros and Cons
+
+- **Pros**
+  - Loading into a workspace is significantly faster than loading into a table without native data types. Casting data is not necessary when loading into a workspace.
+  - A table accessed in a workspace via the [read-only input mapping](https://help.keboola.com/transformations/workspace/#read-only-input-mapping) already has typed columns.
+  - Data types are strictly enforced, ensuring that data in a specific column (like a number column) is consistent with its type.
+- **Cons**
+  - Changing a column type is complicated; see [How to Change Column Types](#changing-types-of-existing-typed-columns).
+  - Keboola does not perform any type conversion during loading. Your data must exactly match the column type in the table in Storage.
+  - Any load of data with incompatible types will fail.
+  - The filtering option in the input mapping section is unavailable for tables with defined data types. If filtering is crucial for your workflow, consider using SQL, Python, or even no-code transformations to filter the data and create a new filtered table.
+
+
+### How to Create a Typed Table Based on a Non-typed Table
+
+For example, we have a non-typed table `non_typed_table` with the following definition:
+
+{: .image-popup}
+![Non-typed table schema](/storage/tables/data-types/non-typed-table-schema.png)
+
+with sample data:
+
+{: .image-popup}
+![Sample data](/storage/tables/data-types/sample-data.png)
+
+To create typed table based on `non_typed_table` create new transformation, choose table input mapping `non_typed_table` and choose table output mapping `typed_table`. The output table must not exist, otherwise it will not become a typed table.
```

Comment:
```suggestion
To create typed table based on `non_typed_table`, create new transformation, choose table input mapping `non_typed_table` (or you can rely on [read-only input mapping](https://help.keboola.com/transformations/#read-only-input-mapping)) and choose table output mapping `typed_table`. The output table must not exist, otherwise it will not become a typed table.
```